### PR TITLE
MMDP-267 Refactor codebase

### PR DESCRIPTION
**Pull request by p8ul**
**What does this PR do?**

	Refactor codebase

**Description of Task to be completed?**

- add air-bnb eslint configuration
- integrate the project with redux-saga
- rename users model to user model
- add package-json script to run/build the client app

**How should this be manually tested?**

Once this PR is merged, we can now run `yarn run client:build` or `yarn run client:watch` to run the client app

**What are the relevant Jira issues?**

[MMDP-267](https://viisaus.atlassian.net/browse/MMDP-267)

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ keystone.init({
   'auto update': true,
   'session': true,
   'auth': true,
-  'mongo': process.env.MONGODB_URL || 'mongodb://127.0.0.1/mmdp-cms',
+  'mongo': 'mongodb://localhost/keystonereactcms',
```
Comment (Alameen688) [inline server/index.js]: I see the initial value of the mongo property has been changed. The environment variable is quite important for production when the username and password of the database will be included as part of the database connection URI. For example: `mongodb://<dbuser>:<dbpassword>@ds259499.mlab.com:59499/employees`.

Also, I feel we shouldn't use the names of the technologies used as our database name as it isn't descriptive of what the database is used for.
```suggestion
  'mongo': 'mongodb://localhost/<another-name>',
```
Kindly help look into the database name and adding of support for the database URL environment variable. Well-done.

Comment (p8ul) [reply server/index.js]: Thank you for the feedback. I'll refactor that code.

Comment (Alameen688) [reply server/index.js]: Alright. You are welcome.

Comment (p8ul) [reply server/index.js]: I have refactored the code.

File: client/containers/login/index.jsx [hunk 2]
```diff
@@ -7,26 +7,29 @@ import { loginLoading } from '../../store/actions/auth/login';
 
 export class Login extends Component {
```
Comment (shemogumbe) [inline client/containers/login/index.jsx]: I am of the opinion that we do not use the extension `.jsx`, let us stick with `.js` as babel would still transpile these and we will also achieve consistency

File: server/index.js (hunk 1 above)
Comment (shemogumbe) [reply server/index.js]: So as we do not create a precedence of assumptions, the db path and the cookie secret should strictly be in the .env and picked as env variables

File: client/containers/login/index.jsx (hunk 2 above)
Comment (p8ul) [reply client/containers/login/index.jsx]: Thanks for the feedback. Working on that

File: server/index.js (hunk 1 above)
Comment (p8ul) [reply server/index.js]: Thank you for the request. I have added a .env_sample, that will guide the team on environment variables needed for the app to run normally.

**Pull request merged**